src/net/rpc.rs

```diff
@@ -136,13 +140,15 @@ impl PeerInfoData {
 }
 
 impl ConversationHttp {
```

Comment:
Just to clarify -- is the intention for ConversationHttp to be used for both the peer network and the JSON rpc interface for things like getting account data, map data, etc. ?

Reply:
Yes -- the `ConversationHttp` struct encapsulates all the state needed to both send and reply to HTTP requests.  It's lifetime is tied to the lifetime of the underlying TCP socket in `HttpPeer` (in `src/net/server.rs`), which instantiates a new `ConversationHttp` for each new socket it opens or accepts.

The code that will handle things like getting accounts, map data, and even running read-only contracts would probably look a lot like the code for `handle_getinfo()` or `handle_getneighbors()` -- it will be an associated function.

Reply:
Yep -- that's what I assumed, and that's what I've been doing so far in implementing #970

Reply:
Gotcha.  For that PR, I take you'll also be making new type variants for `HttpRequestType` and `HttpResponseType`, and adding their requisite parsing and serializing code to `src/net/http.rs`?

Reply:
Yep